cpp/src/lists/drop_list_duplicates.cu

```diff
@@ -22,74 +22,112 @@
 #include <cudf/detail/nvtx/ranges.hpp>
 #include <cudf/lists/detail/sorting.hpp>
 #include <cudf/lists/drop_list_duplicates.hpp>
-#include <cudf/table/row_operators.cuh>
+#include <cudf/utilities/type_dispatcher.hpp>
 
 #include <rmm/cuda_stream_view.hpp>
 #include <rmm/exec_policy.hpp>
 
 #include <thrust/binary_search.h>
+#include <thrust/logical.h>
 #include <thrust/transform.h>
 
 namespace cudf {
 namespace lists {
 namespace detail {
 namespace {
+template <typename Type>
+struct has_negative_nans {
+  column_device_view const d_entries;
+  bool const has_nulls;
+
+  __device__ Type operator()(size_type idx) const noexcept
+  {
+    if (has_nulls && d_entries.is_null(idx)) { return false; }
```

Comment:
If you know the column is nullable, you can skip the check for the `bitmask == nullptr` inside `is_null`. 
```suggestion
    if (has_nulls && d_entries.is_null_nocheck(idx)) { return false; }
```